Sources/Decoder.swift

```diff
@@ -157,6 +157,9 @@ public struct Decoder {
                 
                 for subJSON in jsonArray {
                     guard let model = T(json: subJSON) else {
+                        if T.self is ArrayPartiallyDecodable.Type {
```

Comment:
Encoding relies on the implementation of `Encodable` protocol so, it doesn't matter since any type of object implementing the protocol would be able to be Encoded.

I think this logic applies to Decode since it is an external data (JSON) we don't control, but when encoding you rely on the type system and you have proper objects in place.

I was trying to get some scenario when Encoding partially would be useful the same way Decode is but nothing pops in...

If you have some ideas please let me know.